contracts/cw-govmod-single/src/proposal.rs

```diff
@@ -130,10 +114,115 @@ impl Proposal {
                     // `token_price * quorum` as opposed to the
                     // 'desired' `token_price * threshold`.
                     let options = self.votes.total() - self.votes.abstain;
-                    votes_meet_threshold(vote_count, options, threshold)
+                    votes_meet_threshold(self.votes.yes, options, threshold)
                 } else {
                     let options = self.total_power - self.votes.abstain;
-                    votes_meet_threshold(vote_count, options, threshold)
+                    votes_meet_threshold(self.votes.yes, options, threshold)
+                }
+            }
+        }
+    }
+
+    /// As above for the passed check, used to check if a proposal is
+    /// already rejected.
+    pub fn is_rejected(&self, block: &BlockInfo) -> bool {
+        match self.threshold {
+            Threshold::AbsolutePercentage {
+                percentage: percentage_needed,
+            } => {
+                let options = self.total_power - self.votes.abstain;
+
+                // If there is a 100% passing threshold..
+                if percentage_needed == Decimal::percent(100) {
+                    if options == Uint128::zero() {
+                        // and there are no possible votes (zero
+                        // voting power or all abstain), then this
+                        // proposal has been rejected.
+                        return true;
+                    } else {
+                        // and there are possible votes, then this is
+                        // rejected if there is a single no vote.
+                        //
+                        // We need this check becuase otherwise when
+                        // we invert the threshold (`Decimal::one() -
+                        // threshold`) we get a 0% requirement for no
+                        // votes. Zero no votes do indeed meet a 0%
+                        // threshold.
+                        return self.votes.no >= Uint128::new(1);
+                    }
+                }
+
+                // We need to invert the threshold, when nos exceed
+                // this inverted threshold we know the yes can never
+                // reach the threshold.
+                votes_meet_threshold(self.votes.no, options, Decimal::one() - percentage_needed)
+            }
+            Threshold::ThresholdQuorum { threshold, quorum } => {
+                match (
```

Comment:
@Callum-A you might consider taking a look at the logic here. I've made some modifications to your changes to handle the case where all votes are abstain or there is a 100% passing threshold. You might consider moving this behavior over to the other PR.

Reply:
I've also switched to using a match statement and checking conditions accordingly. This seems more clear as it makes explicit that we cover all four cases of expiration / quorum being met.